servatrice/src/serversocketinterface.cpp

```diff
@@ -747,6 +749,54 @@ Response::ResponseCode ServerSocketInterface::cmdReplayDeleteMatch(const Command
 // MODERATOR FUNCTIONS.
 // May be called by admins and moderators. Permission is checked by the calling function.
 
+Response::ResponseCode ServerSocketInterface::cmdGetBanHistory(const Command_GetBanHistory &cmd, ResponseContainer &rc)
+{
+    if (!sqlInterface->checkSql())
+        return Response::RespInternalError;
+
+    QString userName = QString::fromStdString(cmd.user_name());
+
+    //lookup users ban history from database
+    QSqlQuery *query = sqlInterface->prepareQuery("select id_admin,time_from,minutes,reason,visible_reason from {prefix}_bans where user_name = :user_name");
+    query->bindValue(":user_name", userName);
+    sqlInterface->execSqlQuery(query);
```

Comment:
The query gets executed twice: here and in the next line (with proper error checking). Remove this one maybe?